packages/twenty-server/src/database/commands/0-22-update-message-channel-sync-stage-enum.command.ts

```diff
@@ -0,0 +1,227 @@
+    return value;
+  }
+
+  async run(
+    _passedParam: string[],
+    options: UpdateMessageChannelSyncStageEnumCommandOptions,
+  ): Promise<void> {
+    let workspaceIds: string[] = [];
+
+    if (options.workspaceId) {
+      workspaceIds = [options.workspaceId];
+    } else {
+      workspaceIds = (await this.workspaceRepository.find()).map(
+        (workspace) => workspace.id,
+      );
+    }
+
+    if (!workspaceIds.length) {
+      this.logger.log(chalk.yellow('No workspace found'));
+
+      return;
+    } else {
+      this.logger.log(
+        chalk.green(`Running command on ${workspaceIds.length} workspaces`),
+      );
+    }
+
+    for (const workspaceId of workspaceIds) {
+      const dataSourceMetadatas =
+        await this.dataSourceService.getDataSourcesMetadataFromWorkspaceId(
+          workspaceId,
+        );
+
+      for (const dataSourceMetadata of dataSourceMetadatas) {
+        const workspaceDataSource =
+          await this.typeORMService.connectToDataSource(dataSourceMetadata);
+
+        if (workspaceDataSource) {
+          const queryRunner = workspaceDataSource.createQueryRunner();
+
+          await queryRunner.connect();
+          await queryRunner.startTransaction();
+
+          try {
+            await queryRunner.query(
+              `ALTER TYPE "${dataSourceMetadata.schema}"."messageChannel_syncStage_enum" RENAME TO "messageChannel_syncStage_enum_old"`,
+            );
+            await queryRunner.query(
+              `CREATE TYPE "${dataSourceMetadata.schema}"."messageChannel_syncStage_enum" AS ENUM (
+              'FULL_MESSAGE_LIST_FETCH_PENDING',
+              'PARTIAL_MESSAGE_LIST_FETCH_PENDING',
+              'MESSAGE_LIST_FETCH_ONGOING',
+              'MESSAGES_IMPORT_PENDING',
+              'MESSAGES_IMPORT_ONGOING',
+              'FAILED'
+            )`,
+            );
+
+            await queryRunner.query(
+              `ALTER TABLE "${dataSourceMetadata.schema}"."messageChannel" ALTER COLUMN "syncStage" DROP DEFAULT`,
+            );
+            await queryRunner.query(
+              `ALTER TABLE "${dataSourceMetadata.schema}"."messageChannel" ALTER COLUMN "syncStage" TYPE text`,
+            );
+
+            await queryRunner.query(
+              `ALTER TABLE "${dataSourceMetadata.schema}"."messageChannel" ALTER COLUMN "syncStage" TYPE "${dataSourceMetadata.schema}"."messageChannel_syncStage_enum" USING "syncStage"::text::"${dataSourceMetadata.schema}"."messageChannel_syncStage_enum"`,
+            );
+
+            await queryRunner.query(
+              `DROP TYPE "${dataSourceMetadata.schema}"."messageChannel_syncStage_enum_old"`,
+            );
+            await queryRunner.commitTransaction();
+          } catch (error) {
+            await queryRunner.rollbackTransaction();
+            this.logger.log(
+              chalk.red(`Running command on workspace ${workspaceId} failed`),
+            );
+            throw error;
```

Comment:
so if it fails for any workspace, the command will stop altogether. Don't we want to continue to the next workspace instead? it's what we do elsewhere